controllers/resources/manager.go

```diff
@@ -206,17 +207,11 @@ func (m *manager) UpdateRemediationCR(remediationCR *unstructured.Unstructured)
 	return m.Update(m.ctx, remediationCR)
 }
 
-func (m *manager) ListRemediationCRs(nhc *remediationv1alpha1.NodeHealthCheck, remediationCRFilter func(r unstructured.Unstructured) bool) ([]unstructured.Unstructured, error) {
+func (m *manager) ListRemediationCRs(remediationTemplates []*corev1.ObjectReference, remediationCRFilter func(r unstructured.Unstructured) bool) ([]unstructured.Unstructured, error) {
 	// gather all GVKs
-	gvks := make([]schema.GroupVersionKind, 0)
-	if nhc.Spec.RemediationTemplate != nil {
-		// the classic one
-		gvks = append(gvks, nhc.Spec.RemediationTemplate.GroupVersionKind())
-	} else {
-		// the escalating ones
-		for _, escRem := range nhc.Spec.EscalatingRemediations {
-			gvks = append(gvks, escRem.RemediationTemplate.GroupVersionKind())
-		}
+	gvks := make([]schema.GroupVersionKind, len(remediationTemplates))
```

Comment:
NIT: I am not sure it is possible, from the calls to `ListRemediationCRs`, but can't it be that `remediationTemplates` is nil and ` len(remediationTemplates)` would return an error?

Reply:
why would it return an error?

Reply:
>why would it return an error?

I thought that if `remediationTemplates` is nil, not sure if it is feasible in practice, then trying to get a length of nil would return an error. Looking again, it is not a bug/error, since a nil slice has a length and capacity of 0.